packages/identity/src/identity/secp256k1.ts

```diff
@@ -0,0 +1,149 @@
+
+declare type PublicKeyHex = string;
+declare type SecretKeyHex = string;
+export declare type JsonableSecp256k1Identity = [PublicKeyHex, SecretKeyHex];
+
+export class Secp256k1PublicKey implements PublicKey {
+  public static from(key: PublicKey): Secp256k1PublicKey {
+    return this.fromDer(key.toDer());
+  }
+
+  public static fromRaw(rawKey: ArrayBuffer): Secp256k1PublicKey {
+    return new Secp256k1PublicKey(rawKey);
+  }
+
+  public static fromDer(derKey: DerEncodedPublicKey): Secp256k1PublicKey {
+    return new Secp256k1PublicKey(this.derDecode(derKey));
+  }
+
+  private static derEncode(publicKey: ArrayBuffer): DerEncodedPublicKey {
+    return wrapDER(publicKey, SECP256K1_OID).buffer as DerEncodedPublicKey;
+  }
+
+  private static derDecode(key: DerEncodedPublicKey): ArrayBuffer {
+    return unwrapDER(key, SECP256K1_OID);
+  }
+
+  private readonly rawKey: ArrayBuffer;
+
+  private readonly derKey: DerEncodedPublicKey;
+
+  // `fromRaw` and `fromDer` should be used for instantiation, not this constructor.
+  private constructor(key: ArrayBuffer) {
+    key.byteLength;
+    this.rawKey = key;
+    this.derKey = Secp256k1PublicKey.derEncode(key);
+  }
+
+  public toDer(): DerEncodedPublicKey {
+    return this.derKey;
+  }
+
+  public toRaw(): ArrayBuffer {
+    return this.rawKey;
+  }
+}
+
+export class Secp256k1KeyIdentity extends SignIdentity {
+  public static generate(seed?: Uint8Array): Secp256k1KeyIdentity {
+    if (seed && seed.length !== 32) {
+      throw new Error('Secp256k1 Seed needs to be 32 bytes long.');
+    }
+
+    let privateKey = seed || new Uint8Array(randomBytes(32));
+
+    while (!Secp256k1.privateKeyVerify(privateKey)) {
+      privateKey = new Uint8Array(randomBytes(32));
+    }
+    const publicKeyRaw = Secp256k1.publicKeyCreate(privateKey, false);
+
+    const publicKey = Secp256k1PublicKey.fromRaw(publicKeyRaw);
+    return new this(publicKey, privateKey);
+  }
+
+  public static fromParsedJson(obj: JsonableSecp256k1Identity): Secp256k1KeyIdentity {
+    const [publicKeyRaw, privateKeyRaw] = obj;
+    return new Secp256k1KeyIdentity(
+      Secp256k1PublicKey.fromRaw(fromHexString(publicKeyRaw) as DerEncodedPublicKey),
+      fromHexString(privateKeyRaw),
+    );
+  }
+
+  public static fromJSON(json: string): Secp256k1KeyIdentity {
+    const parsed = JSON.parse(json);
+    if (Array.isArray(parsed)) {
+      if (typeof parsed[0] === 'string' && typeof parsed[1] === 'string') {
+        return this.fromParsedJson([parsed[0], parsed[1]]);
+      }
+      throw new Error('Deserialization error: JSON must have at least 2 items.');
+    } else if (typeof parsed === 'object' && parsed !== null) {
+      throw new Error('Deprecated JSON format for Ed25519 keys.');
+    }
+    throw new Error(`Deserialization error: Invalid JSON type for string: ${JSON.stringify(json)}`);
+  }
+
+  public static fromKeyPair(publicKey: ArrayBuffer, privateKey: ArrayBuffer): Secp256k1KeyIdentity {
```

Comment:
nit: maybe add documentation to this method stating that it is the responsibility of the caller to generate the private key in a secure way?